File: ДКР 1/Pascal/1.pas
uses Math;
var
  x, y: Real;
begin
  ReadLn(x);
  if x < -9 then
  begin
    if x <> 0 then
      y := Sign(x) * Power(Abs(x), 1/3) * Power(Abs(x), 0.1 * x) + Power(x, 3) / Log10(Abs(x))
    else
      y := 0;
  end
  else if (x >= -9) and (x < 1) then
  begin
    if x > 0 then
      y := Power(x, 2) * Ln(x)
    else
      y := 0;
  end
  else if (x >= 1) and (x < 4) then
    y := Exp(x) * (-x)
  else
    y := Log10(x) * Sin(x);
  WriteLn('x = ', x:0:2, '; y = ', y:0:2);
end.

File: ДКР 1/Python/1.py
import math

x = float(input())
if x < -9:
    if x != 0:
        y = math.copysign(abs(x) ** (1/3), x) * (abs(x) ** (0.1 * x)) + (x ** 3) / math.log10(abs(x))
    else:
        y = 0
elif -9 <= x < 1:
    if x > 0:
        y = (x ** 2) * math.log(x)
    else:
        y = 0
elif 1 <= x < 4:
    y = math.exp(x) * (-x)
else:
    y = math.log10(x) * math.sin(x)
print(f"x = {x:.2f}; y = {y:.2f}")

